packages/eas-build-job/src/job.ts

```diff
@@ -4,17 +4,17 @@ import { Platform } from './common';
 import * as Android from './android';
 import * as Ios from './ios';
 
-export type Job = Android.Job | Ios.Job;
+export type BuildJob = Android.Job | Ios.Job;
 
-export const JobSchema = Joi.object<Job>({
+export const JobSchema = Joi.object<BuildJob>({
   platform: Joi.string()
     .valid(...Object.values(Platform))
     .required(),
 })
   .when(Joi.object({ platform: Platform.ANDROID }).unknown(), { then: Android.JobSchema })
   .when(Joi.object({ platform: Platform.IOS }).unknown(), { then: Ios.JobSchema });
```

Comment:
Maybe if we distinguish between `BuildJob` and `GenericJob` based on the `type` field we can have one conditional Joi/Zed schema to validate Jobs? I'm not sure if there is any benefit to it. Just thinking out loud.

Reply:
I would love to have one Zod schema to validate everything! I didn't want to embark on a journey to rewrite all (or at least Job-related) Joi schemas to Zod. If I remember correctly it touches `www` too and is in general a bigger project to do.